jdaviz/app.py

```diff
@@ -542,6 +543,51 @@ def get_subsets_from_viewer(self, viewer_reference, data_label=None, subset_type
                                          cls=None)
         regions = {}
 
+        def _get_all_subregions(mask, spec_axis_data):
+            """
+            Return all subregions within a subset.
+
+            Parameters
+            ----------
+            mask : list
+                List of indices in spec_axis_data that are part of the subset.
+            spec_axis_data : list
+                List of spectral axis values.
+            Returns
+            -------
+            combined_spec_region : `~specutils.SpectralRegion`
+                SpectralRegion object containing all subregions of the subset.
+            """
+            if len(mask) == 0:
+                # Mask should only be 0 if ApplyROI is used to incorrectly
+                # create subsets via the API
+                raise ValueError
+
+            current_edge = 0
+            combined_spec_region = None
+            for index in range(1, len(mask)):
+                # Find spot where mask == True is for a different region of the subset
+                # i.e. mask = [0, 1, 4, 5]
+                # mask[2] != mask[1] + 1
+                if mask[index] != mask[index - 1] + 1:
+                    subset_region = spec_axis_data[mask[current_edge]: mask[index] - 1]
```

Comment:
I suspect this is where a slight tweak is needed.

Reply:
Good catch!!